library/src/test/java/com/datatorrent/lib/join/AntiJoinOperatorTest.java

```diff
@@ -0,0 +1,124 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package com.datatorrent.lib.join;
+
+import com.datatorrent.lib.streamquery.condition.Condition;
+import com.datatorrent.lib.streamquery.condition.JoinColumnEqualCondition;
+import com.datatorrent.lib.streamquery.index.ColumnIndex;
+import com.datatorrent.lib.testbench.CollectorTestSink;
+import com.datatorrent.lib.util.TestUtils;
+
+import org.junit.Test;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.HashMap;
+
+/**
+ *
+ * Functional test for {@link AntiJoinOperator }.
+ *
+ */
+public class AntiJoinOperatorTest
+{
+  //@SuppressWarnings({"rawtypes", "unchecked"})
+  @Test
+  public void testSqlSelect()
+  {
+    Logger LOG = LoggerFactory.getLogger(AntiJoinOperator.class);
+
+    // create operator
+    AntiJoinOperator oper = new AntiJoinOperator();
+    CollectorTestSink sink = new CollectorTestSink();
+    TestUtils.setSink(oper.outport, sink);
+
+    // set column join condition
+    Condition cond = new JoinColumnEqualCondition("a", "a");
+    oper.setJoinCondition(cond);
+
+    // add columns
+    oper.selectTable1Column(new ColumnIndex("b", null));
+    oper.selectTable1Column(new ColumnIndex("c", null));
+
+    oper.setup(null);
+    HashMap<String, Object> tuple = new HashMap<String, Object>();
+
+    // test 1, positive result
+    oper.beginWindow(1);
+    tuple.put("a", 0);
+    tuple.put("b", 1);
+    tuple.put("c", 2);
+    oper.inport1.process(tuple);
+
+    tuple = new HashMap<String, Object>();
+    tuple.put("a", 1);
+    tuple.put("b", 3);
+    tuple.put("c", 4);
+    oper.inport1.process(tuple);
+
+    tuple = new HashMap<String, Object>();
+    tuple.put("a", 0);
+    tuple.put("b", 7);
+    tuple.put("c", 8);
+    oper.inport2.process(tuple);
+
+    tuple = new HashMap<String, Object>();
+    tuple.put("a", 2);
+    tuple.put("b", 5);
+    tuple.put("c", 6);
+    oper.inport2.process(tuple);
+
+    oper.endWindow();
+
+    // expected anti-joined result: {b=3, c=4}
+    LOG.info("anti-join result: " + sink.collectedTuples.toString());
```

Comment:
This needs to be an assertion instead of logging. Say in future with changes to the operator, a bug is introduced. With assertions we can verifying that the existing functionality is not broken. By just logging we will not be able to do so.